crates/compression/src/decompress.rs

```diff
@@ -68,12 +70,27 @@ where
         db,
     };
 
-    let transactions = <Vec<Transaction> as DecompressibleBy<_>>::decompress_with(
+    let mut transactions = <Vec<Transaction> as DecompressibleBy<_>>::decompress_with(
         block.transactions(),
         &ctx,
     )
     .await?;
 
+    let transaction_count = transactions.len();
+
+    // patch mint transaction
+    for tx in transactions.iter_mut() {
+        if let Transaction::Mint(mint) = tx {
+            let tx_pointer = mint.tx_pointer_mut();
+            // this will break if we have multiple mints
```

Comment:
I would love to know if we have any concrete plans to support multiple mints. The current fault proof design assumes (and ensures) that blocks should have exactly one mint transaction.

Reply:
I don't think that we will have multiple